custom_components/opensprinkler/switch.py

```diff
@@ -84,6 +85,86 @@ def icon(self) -> str:
 
         return "mdi:barley-off"
 
+    @property
+    def device_state_attributes(self):
+        controller = self._controller
+        attributes = {"opensprinkler_type": "controller"}
+        for attr in [
+            "firmware_version",
+            "hardware_version",
+            "hardware_type",
+            "last_run_station",
+            "last_run_program",
+            "last_run_duration",
+            "rain_sensor_active",
+            "sensor_1_enabled",
+            "sensor_2_enabled",
+            "rain_sensor_enabled",
+            "flow_sensor_enabled",
+            "soil_sensor_enabled",
+            "last_weather_call",
+            "last_successfull_weather_call",
+            "last_weather_call_error",
+            "last_weather_call_error_name",
+            "last_reboot_time",
+            "last_reboot_cause",
+            "last_reboot_cause_name",
+        ]:
+            try:
+                attributes[attr] = getattr(controller, attr)
+            except:
+                pass
+
+        for attr in [
+            "last_weather_call",
+            "last_successfull_weather_call",
+            "last_reboot_time",
+        ]:
+            iso_attr = attr + "_iso"
+            timestamp = getattr(controller, attr)
+            if not timestamp:
+                attributes[attr] = None
+                attributes[iso_attr] = None
+            else:
+                attributes[iso_attr] = utc_from_timestamp(timestamp).isoformat()
```

Comment:
Do we need both time formats? can we settle with one?

Reply:
Probably just iso I’d assume as the best option for integrating with hass?